sao_paulo/src/main/java/org/eqasim/sao_paulo/mode_choice/utilities/predictors/SaoPauloTaxiPredictor.java

```diff
@@ -3,39 +3,55 @@
 import java.util.List;
 
 import org.eqasim.core.simulation.mode_choice.cost.CostModel;
+import org.eqasim.core.simulation.mode_choice.parameters.ModeParameters;
 import org.eqasim.core.simulation.mode_choice.utilities.predictors.CachedVariablePredictor;
 import org.eqasim.core.simulation.mode_choice.utilities.predictors.PredictorUtils;
 import org.eqasim.sao_paulo.mode_choice.utilities.variables.TaxiVariables;
+import org.matsim.api.core.v01.TransportMode;
 import org.matsim.api.core.v01.population.Leg;
 import org.matsim.api.core.v01.population.Person;
 import org.matsim.api.core.v01.population.PlanElement;
 import org.matsim.contribs.discrete_mode_choice.model.DiscreteModeChoiceTrip;
+import org.matsim.core.router.TripStructureUtils;
 
+import com.google.common.base.Verify;
 import com.google.inject.Inject;
 import com.google.inject.name.Named;
 
 public class SaoPauloTaxiPredictor extends CachedVariablePredictor<TaxiVariables> {
 	private final CostModel costModel;
+	private final ModeParameters parameters;
 
 	@Inject
-	public SaoPauloTaxiPredictor(@Named("taxi") CostModel costModel) {
+	public SaoPauloTaxiPredictor(ModeParameters parameters, @Named("taxi") CostModel costModel) {
 		this.costModel = costModel;
+		this.parameters = parameters;

```

Comment:
ModeParameters object is for now not used, please remove it (see below my further comments).